deploy-tron/src/handlers/deploy.ts

```diff
@@ -25,29 +26,26 @@ export const deploy = async (context: Context): Promise<void> => {
   // @ts-ignore
   const allowsMultipleDeploysToEnv = config.environmentsThatAllowConcurrentDeploys.findIndex(env => env === environment) > -1;
   
-  const pendingDeploymentsExist = await isTherePendingDeploymentForEnvironment(context, ref, environment, repo, owner);
+  const pendingDeployments = await isTherePendingDeploymentForEnvironment(context, ref, environment, repo, owner);
```

Comment:
Just dusting off my ts knowledge, do these awaits pause the script execution until they're resolved?  Just making sure we don't get a race condition on the check a few lines down

Reply:
await and async are native javscript features. It is just syntax sugar on top of Promises. Just makes it read more like synchronous code

Reply:
the variable is the returned Promise. No race condition :)